external-crates/move/documentation/book/src/control-flow/pattern-matching.md

```diff
@@ -0,0 +1,567 @@
+# Pattern Matching
+
+A `match` expression is a powerful control structure that allows you to compare a value against a
+series of patterns and then execute code based on which pattern matches first. Patterns can be
+anything from simple literals to complex structures. As opposed to `if` expressions, which require a
+`bool`ean expression, a `match` expression requires a value to be matched against a series of
+patterns.
+
+A `match` expression can match either by value, immutable reference, or mutable reference.
+
+A pattern is matched by a value if the value is equal to the pattern, and where variables and
+wildcards (e.g., `x`, `y`, `_`, or `..`) are "equal" to anything.
+
+For example:
+
+```move
+fun run(x: u64): u64 {
+    match (x) {
+        1 => 2,
+        2 => 3,
+        x => x,
+    }
+}
+
+run(1); // returns 2
+run(2); // returns 3
+run(3); // returns 3
+run(0); // returns 0
+```
```

Comment:
I'm in favor of a "flashy" first example. If I end up here, I want the csug description: show me all the usage in 1-2 code examples with all the behavior, then take the time to explain. That way if I'm looking for something quick, it's right here at the top of the file.

Reply:
I like it, let me think about what a good example like that might be up here :)